.github/workflows/docs-ci.yml

```diff
@@ -1,39 +1,55 @@
-# This workflow will build and deploy docs to GitHub Pages whenever something is pushed to the default branch
+name: Build docs
 
-name: Docs
-
-on: push # but only to default branch, see if below
+on:
+  push:
+    branches: [ "0.9.x" ]
+  tag:
+  workflow_dispatch:
 
+# Gives the workflow permissions to clone the repo and create a page deployment
 permissions:
-  contents: write
+  id-token: write
+  pages: write
 
+# This workflow updates the versioned docs on dokka.kord.dev using the following strategy
+# 1) pull cache to docs-cache
+# 2) Call dokkaHtmlMultiModule for current version
+#   - if this is a release build move current docs into docs-cache/$version
+#   - otherwise move it into docs-cache/current
+# 3) Build GitHub Pages artifact
+#   - Move all folders not named "current" into docs/<folder>
+#   - Move current into docs/
+# 4) Trigger deploy to GitHub Actions
 jobs:
-  docs:
-    name: Build and deploy docs
+  build-job:
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
     runs-on: ubuntu-latest
-    if: github.ref == format('refs/heads/{0}', github.event.repository.default_branch)
-    concurrency: # Allow one concurrent deployment
-      group: pages
-      cancel-in-progress: true
-    env:
-      GITHUB_TAG_NAME: ${{ github.event.release.tag_name }}
-      GITHUB_BRANCH_NAME: ${{ github.ref }}
+    GITHUB_TAG_NAME: ${{ github.event.release.tag_name }}
+    GITHUB_BRANCH_NAME: ${{ github.ref }}
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout repository
+        uses: actions/checkout@v3
       - name: Set up JDK
         uses: actions/setup-java@v3
         with:
           distribution: temurin
           java-version: 8
+      - uses: actions/cache@v3
+        with:
+          path: docs-cache
+          key: docs-cache
       - name: Build docs with Gradle
         uses: gradle/gradle-build-action@v2
         with:
-          arguments: --stacktrace --info dokkaHtmlMultiModule
-      - name: Deploy docs to GitHub Pages
-        uses: JamesIves/github-pages-deploy-action@v4
+          arguments: :produceDocs
```

Comment:
where does this task come from?

Reply:
it's called `:produceDocs` the `:` means it's from the root project, so it's in the root `build.gradle.kts`

Reply:
did you commit it?